File: objc/Strategy/Strategy/Post.m
#import "Post.h"

@implementation Post

-(instancetype)initWith:(NSString*)text
              likeCount:(int)likeCount
               sadCount:(int)sadCount
             laughCount:(int)laughCount {
    if(self = [super init]) {
        _text = text;
        _likeCount = likeCount;
        _sadCount = sadCount;
        _laughCount = laughCount;
    }
    return self;
}

@end

File: swift/Strategy/Strategy/Post.swift
import Foundation

class Post {
    var text: String?
    var likeCount: Int = 0
    var sadCount: Int = 0
    var laughCount: Int = 0
    
    init(_ text: String) {
        self.text = text
    }
    
    convenience init(_ text: String, _ likeCount: Int, _ sadCount: Int, _ laughCount: Int) {
        self.init(text)
        self.likeCount = likeCount
        self.sadCount = sadCount
        self.laughCount = laughCount
    }
}
